dask/dataframe/io.py

```diff
@@ -637,57 +640,30 @@ def to_castra(df, fn=None, categories=None, sorted_index_column=None,
         return delayed([Delayed(key, [dsk]) for key in keys])
 
 
-def to_csv(df, filename, name_function=None, compression=None, get=None, compute=True, **kwargs):
-    if compression:
-        raise NotImplementedError("Writing compressed csv files not supported")
-    name = 'to-csv-' + uuid.uuid1().hex
+def to_csv(df, filename, name_function=None, compression=None, compute=True,
+           **kwargs):
 
-    kwargs2 = kwargs.copy()
-
-    if name_function is None:
-        name_function = build_name_function(df.npartitions - 1)
-
-    if '*' in filename:
-        if filename.count('*') > 1:
-            raise ValueError("A maximum of one asterisk is accepted in filename")
-
-        if 'mode' in kwargs and kwargs['mode'] != 'w':
-            raise ValueError("to_csv does not support writing to multiple files in append mode, "
-                             "please specify mode='w'")
-
-        formatted_names = [name_function(i) for i in range(df.npartitions)]
-        if formatted_names != sorted(formatted_names):
-            warn("To preserve order between partitions name_function "
-                 "must preserve the order of its input")
-
-        single_file = False
-    else:
-        kwargs2.update({'mode': 'a', 'header': False})
-        single_file = True
-
-    dsk = dict()
-    dsk[(name, 0)] = (lambda df, fn, kwargs: df.to_csv(fn, **kwargs),
-                        (df._name, 0), filename.replace('*', name_function(0)), kwargs)
-
-    for i in range(1, df.npartitions):
-        filename_i = filename.replace('*', name_function(i))
-
-        task = (lambda df, fn, kwargs: df.to_csv(fn, **kwargs),
-                 (df._name, i), filename_i, kwargs2)
-        if single_file:
-            task = (_link, (name, i - 1), task)
-        dsk[(name, i)] = task
-
-    dsk = merge(dsk, df.dask)
-    if single_file:
-        keys = [(name, df.npartitions - 1)]
-    else:
-        keys = [(name, i) for i in range(df.npartitions)]
+    def func(df, **kwargs):
+        if PY2:
+            out = io.BytesIO()
+        else:
+            out = io.StringIO()
+        df.to_csv(out, **kwargs)
+        out.seek(0)
+        if PY2:
+            return out.getvalue()
+        return out.getvalue().encode(encoding)
```

Comment:
I don't see any way to do this better than you've already done it here.  This does appear to be a bit annoying though.  I see your point.